### PR TITLE
storage: Move common code in helper functions, closes #730

### LINKED ISSUE #730: storage: refactor pkg/storage
**Describe the bug**

Currently, we have modeled different storage backends as completely independent modules. However, most of the code is identical (investigate if this is true), so perhaps refactor as common code and backend-specific code.

### PR DESCRIPTION
**Pull request by peusebiu**
Signed-off-by: Petu Eusebiu <peusebiu@cisco.com>

Cleanup storage code, moved some common code into helper.go

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/storage/common.go [hunk 1]
```diff
@@ -0,0 +1,413 @@
...
+
+			return "", zerr.ErrBadManifest
+		}
+	}
+
+	return bodyDigest, err
+}
+
+/* CheckIfIndexNeedsUpdate verifies if an index needs to be updated given a new manifest descriptor.
+Returns whether or not index needs update, in the later case it will also return the previous digest. */
```
Comment (alexstan12) [inline pkg/storage/common.go:148]: typo, maybe use "in the latter case it will also return"

**Comment by alexstan12** [comment]
lgtm.

**Comment by rchincha** [comment]
@peusebiu move pkg/storage/local.go into its own pkg/storage/local/local.go now so everything is symmetric.

File: pkg/storage/common.go (hunk 1 above)
Comment (peusebiu) [reply pkg/storage/common.go]: done, thanks

**Comment by peusebiu** [comment]
> @peusebiu move pkg/storage/local.go into its own pkg/storage/local/local.go now so everything is symmetric. Otherwise lgtm

done, thanks

**Pull request merged**